src/libraries/Microsoft.Extensions.Configuration.FileExtensions/src/FileConfigurationProvider.cs

```diff
@@ -162,11 +164,6 @@ private void HandleException(ExceptionDispatchInfo info)
         protected virtual void Dispose(bool disposing)
         {
             _changeTokenRegistration?.Dispose();
-
-            if (Source.OwnsFileProvider)
-            {
-                (Source.FileProvider as IDisposable)?.Dispose();
```

Comment:
`Provider.Dispose` does not need to dispose the source anymore, the caller can access `ConfigurationSource` and do it when needed